src/Control/Monad/Catch.hs

```diff
@@ -191,13 +191,13 @@ class MonadCatch m => MonadMask m where
   generalBracket
     :: m a
     -- ^ acquire some resource
-    -> (a -> b -> m c)
+    -> (a -> m ignored1)
```

Comment:
Currently, the documentation for `generalBracket` states:

> The former function is provided the acquired value and the inner function's result, and returns a new result value.

I assume this is referring to the function of type `(a -> m ignored1)`, correct? If so, isnt't this documentation now incorrect, as this function no longer returns a result value?